spec/adapter_spec.rb

```diff
@@ -9,6 +9,15 @@
     allow(QuickTravel::Api).to receive(:post).and_return(response)
   end
 
+  context 'when the query needs to be url encoded' do
+    let(:url) { 'http://test.quicktravel.com.au/ blah' }
```

Comment:
Shouldn't we just encode the query or the parameters instead of encoding the whole URL?

The original code name the variable `path`, which could be changed to `url`, and like I said above, we normally do not need/should not encode the whole URL.

Reply:
A way to improve this is to find out where `self.call_and_validate` is called, and change the caller so it passes in a valid URL with the encoded route, query and parameters.

Reply:
call_and_validate actually builds the resulting url so I feel it is the best spot to encode it as well